File: db/format/formats.go
package format

import (
	"github.com/qiniu/uip/db/inf"
)

func RegisterDumpFormat(ext string, create inf.Dump) {
	dumpFormats[ext] = create
}

func RegisterQueryFormat(ext string, create inf.NewQuery) {
	queryFormats[ext] = create
}

func RegisterPackFormat(ext string, create inf.Pack) {
	packFormats[ext] = create
}

func GetDumpFormat(ext string) inf.Dump {
	return dumpFormats[ext]
}

func GetQueryFormat(ext string) inf.NewQuery {
	return queryFormats[ext]
}

func GetPackFormat(ext string) inf.Pack {
	return packFormats[ext]
}

var dumpFormats = map[string]inf.Dump{}

var queryFormats = map[string]inf.NewQuery{}

var packFormats = map[string]inf.Pack{}

File: java/src/main/java/qiniu/uip/db/format/Formats.java
package qiniu.uip.db.format;

import java.util.HashMap;
import java.util.Map;

public final class Formats {
    private static Map<String, QueryBuilder> queryFormats = new HashMap<>();

    private Formats() {
    }

    public static void registerQueryFormat(String ext, QueryBuilder q) {
        queryFormats.put(ext, q);
    }

    public static QueryBuilder getQueryFormat(String ext) {
        return queryFormats.get(ext);
    }
}
